src/nspawn/nspawn.c

```diff
@@ -1074,6 +1074,15 @@ static int parse_argv(int argc, char *argv[]) {
         if (getenv_bool("SYSTEMD_NSPAWN_MOUNT_RW") <= 0)
                 mount_settings |= MOUNT_RO;
 
+        if (arg_userns_mode != USER_NAMESPACE_NO)
+                mount_settings |= MOUNT_USE_USERNS;
+
+        if (arg_private_network)
+                mount_settings |= MOUNT_USE_NETNS;
+
+        if (getenv_bool("SYSTEMD_NSPAWN_USE_NETNS") <= 0)
+                mount_settings |= MOUNT_USE_NETNS;
```

Comment:
so, what precisely is left of `--private-network`'s functionality if you use `$SYSTEMD_NSPAWN_USE_NETNS`? Only the fact that /proc/sys/net remains writable?

So, summarizing your patch set with `$SYSTEMD_NSPAWN_MOUNT_RW` you get all of /proc/, /sys writable, but with `$SYSTEMD_NSPAWN_USE_NETNS` you get only /proc/sys/net writable but everything else still read-only? If so, the latter option is a true subset of the former? If so, maybe this should not be two bool env vars, but instead one that takes an enum? Maybe `$SYSTEMD_API_VFS_WRITABLE=auto|yes|no|network` or so?